ophyd/log.py

```diff
@@ -150,44 +194,70 @@ def set_handler(file=sys.stdout, datefmt='%H:%M:%S', color=True):
         Date format. Default is ``'%H:%M:%S'``.
     color : boolean
         Use ANSI color codes. True by default.
-
+    level : str or int
+        Python logging level, given as string or corresponding integer.
+        Default is 'INFO'.
+    backupCount : int
+        Number of historical log files to keep. Default is 4.
     Returns
     -------
     handler : logging.Handler
         The handler, which has already been added to the 'ophyd' logger.
-
     Examples
     --------
     Log to a file.
-
-    >>> set_handler(file='/tmp/what_is_happening.txt')
-
+    >>> config_ophyd_logging(file='/tmp/what_is_happening.txt')
     Include the date along with the time. (The log messages will always include
     microseconds, which are configured separately, not as part of 'datefmt'.)
-
-    >>> set_handler(datefmt="%Y-%m-%d %H:%M:%S")
-
+    >>> config_ophyd_logging(datefmt="%Y-%m-%d %H:%M:%S")
     Turn off ANSI color codes.
-
-    >>> set_handler(color=False)
+    >>> config_ophyd_logging(color=False)
+    Increase verbosity: show level DEBUG or higher.
+    >>> config_ophyd_logging(level='DEBUG')
     """
     global current_handler
     if isinstance(file, str):
-        handler = logging.FileHandler(file)
+        handler = logging.handlers.TimedRotatingFileHandler(
+            filename=file,
+            when="W0",  # roll over every Monday
+            backupCount=backupCount,  # keep the most recent "backupCount" log files
+        )
     else:
         handler = logging.StreamHandler(file)
+    levelno = validate_level(level)
+    handler.setLevel(levelno)
     if color:
-        format = color_log_format
+        log_format = color_log_format
     else:
-        format = plain_log_format
-    handler.setFormatter(
-        LogFormatter(format, datefmt=datefmt))
+        log_format = plain_log_format
+    handler.setFormatter(LogFormatter(log_format, datefmt=datefmt))
+
     if current_handler in logger.handlers:
         logger.removeHandler(current_handler)
     logger.addHandler(handler)
+
+    if current_handler in control_layer_logger.handlers:
```

Comment:
I would like to use the same handler for all ophyd-related logging and by default the same logging level, presuming the control_layer_logger is a temporary measure. I would prefer to leave it to developers to turn it down with setLevel().

Reply:
That is OK with me. In that case, I don't see any reason to give it a name outside the `ophyd.*` logger namespace. Confining ourselves to `ophyd.*` seems good behavior so that someone wishing to _silence_ our logs only has one top-level logger to touch. I only suggested going outside `ophyd.*` in order to side-step this mechanism, so if we decide not to side-step it let's call the logger `ophyd.control_layer`. It can of course have its level separately set, as you suggest. Also, I think all control layer messages should be DEBUG or maybe a custom unnamed level lower than debug (i.e. 0 < level < 10), up to you.

Reply:
I agree that control layer messages should be DEBUGs.

Reply:
Bump on this too.